Library/Homebrew/os/mac/xcode.rb

```diff
@@ -139,7 +139,8 @@ def uncached_version
       end
 
       def provides_autotools?
-        version < "4.3"
+        # Xcode 2.5's autotools are too old to rely on at this point
+        (version < "4.3") && (version > "2.5")
```

Comment:
That was the intention, yeah - this marks Xcodes 2.5 and older as not providing autotools, so Homebrew uses its own.